### ISSUE TITLE
iSC/Cold chain/GAPS: some asset attributes are not configured/missing from export/import functionality

### ISSUE THREAD

**Issue by marthakeezy** [msupply-foundation/open-msupply#5481]
## What went wrong? 😲

I was playing around with importing/exporting cold chain assets, and noticed some attributes were not included in the import template or when exporting a CSV file -> therefore the user would have to go through each asset manually to update the missing details. 

**The missing ones are:** 

- Store
- Category
- Type
- Warranty Start
- Warranty End
- Current Status
- Reason
- Needs Replacement (Y/N)
- Donor

**Enhancements:** 

- Can lifespan column be placed after Refrigerant Type(s) just as it appears in omSupply? Currently, it's the last column in the template/export file which does not make sense to me
- Temperature Monitoring Device appears twice in the file

<img width="778" alt="Screenshot 2024-11-20 at 10 56 31 AM" src="https://github.com/user-attachments/assets/2ccf2ede-7e54-4449-b0f0-2295549431b2">

- Catalogue Item code appears in export and is editable in import template, however does not appear in omSupply (front-end wise?). Should this info be visible to the users? Possibly needs @adamdewey's input

## Expected behaviour 🤔

If the field is editable/fillable by the user, it should be in the import template & also exported, otherwise the user will have to manually update what's missing

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to cold chain -> assets
2. Create a new asset and fill out all the details (e.g. cold chain location, measurements, donor etc)
3. Export the list of assets - see some columns are missing, no way for the user to export some information
4. When uploading assets, download the template provided - notice the same columns are missing from the template
5. Inconsistency!

## Your environment 🌱

- Open mSupply Version: V2.4.0-RC1
- Legacy mSupply Central Server Version: V7.18.11

- Platform: Google Chrome (browser on Mac)

- Database type: SQLite3

**Comment by jmbrunskill** [comment]
@marthakeezy I agree I think it's a normal, but might get triages as high. We do want to setup a bunch of new assets with GAPS.

**Comment by lache-melvin** [comment]
Closed by #5553

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.6.0-cooldown-build-test`

- [ ] Have some CCE assets
- [ ] Export assets
  - [ ] warranty dates, current status and whether it needs replacing should come through
  - [ ] Properties should not be duplicated
- [ ] Click import asset
  - [ ] Download the import CSV
  - [ ] Fill it out with some new assets (at least one functioning, one with a different status, one that needs replacing)
  - [ ] Import the CSV
  - [ ] Assets import correctly (dates and statuses mapped! - note that if you don't set a status, it will be created as functioning)
  - [ ] Import again to create errors (already exists)
  - [ ] Export the errors CSV
  - [ ] Columns all look correct here
- [ ] Compare all the CSV files, columns should now all be in the same order
- [ ] Check all works on both central server and remote site (central should include store column, remote should not)

**Comment by marthakeezy** [comment]
hey team! just a few important notes i want to bring up after testing this one: 

1. When importing CCEI assets, I accidentally misspelled the word Decommissioned as Decomissioned (with one 'm') and as a result the system did not recognise it as a valid status and overruled it as Functioning. I think this is a huge assumption, and possibly anything that does not match the exact value should be rejected as an error?

<img width="226" alt="Image" src="https://github.com/user-attachments/assets/4042be25-15b0-4516-a8a5-85f6abca7988" />

<img width="217" alt="Image" src="https://github.com/user-attachments/assets/c3afbc43-7baf-41c7-b28c-e882a9f28754" />

2. It is extremely unclear how some cells/columns should be filled out. My areas of confusion are any configurable asset properties, that are clearly defined in omSupply but not in the CSV file - columns such as external dimensions, storage capacity, expected lifespan. The column headers do not provide any guidelines around the format the cells should follow or the units of measurement - is it possible to specify them just like we did for timestamps (adding UTC for clarity)?

Comment from Mark on this: `trickier - as the asset properties are configurable. can't really put anything in the docs more than what we have`

**Template (unclear):** 

<img width="289" alt="Image" src="https://github.com/user-attachments/assets/90b57a3c-8cb2-4735-a579-146e0ca9c92f" />

**omSupply (clearer):** 

<img width="379" alt="Image" src="https://github.com/user-attachments/assets/da46b118-e4cf-4346-924f-3a68b6c6979d" />

3. Scenario: I am uploading assets and run into errors. I export the errors to see what's going on.

When uploading, Needs Replacement column looks like this (TRUE/FALSE): 

<img width="289" alt="Image" src="https://github.com/user-attachments/assets/8424c8cc-5bf7-4a65-bb62-d7ab373babc7" />

When exporting the error file, Needs Replacement looks like this: 

<img width="161" alt="Image" src="https://github.com/user-attachments/assets/ec3a85f6-69cf-4c7f-8390-e651a83f1304" />

Considering importing and exporting is already a big confusing, can we keep the same format/values?

4. Same scenario - I am uploading assets and run into errors. I export the errors to see what's going on.

When uploading, any cells that have date values (e.g. Installation Date) follow the format of DD/MM/YYYY.

<img width="435" alt="Image" src="https://github.com/user-attachments/assets/2f934734-c4f1-4456-a834-9f96e2aa4da5" />

When exporting the error file, the same cells follow a different format - YYYY-MM-DD

<img width="424" alt="Image" src="https://github.com/user-attachments/assets/d0aeaf0f-3e05-41ac-9a97-1f905c63a608" />

Let's say the user keeps working in the same file and misses the change (which is completely realistic because I did) - they upload the same file after fixing the error and are faced with the following message: 

<img width="284" alt="Image" src="https://github.com/user-attachments/assets/1fa0973c-fcbf-4917-ba41-2a1d8fb5c1f7" />

The cells are nullified and they have to fill them out manually again. That is a lot of double-handling, so when exporting the error file, can we change all the date-related cells to follow the format of DD/MM/YYYY please?

Related issues that I found on GitHub: 

- [ ] #6280 
- [ ] #5682
- [ ] #5762

**Comment by lache-melvin** [comment]
Hi @marthakeezy - would you mind instead raising this as a new issue and linking this one for reference. Original issue (the fields are not included) has been implemented. Updates to how we validate the imported data, or communicate what the fields mean, should be triaged separately 🙏

**Comment by marthakeezy** [comment]
Thanks! Raised the following: 

#6317
#6319
#6320
